### ISSUE TITLE
Cilium 502 Bad Gateway on control-planes

### ISSUE THREAD

**Issue by fmunteanu** [axivo/k3s-cluster#118]
### Existing Issues

- [X] I have searched the existing issues

### What Happened

Cilium reports `502 Bad Gateway` errors:

```
# cilium status
    /¯¯\
 /¯¯\__/¯¯\    Cilium:             8 errors
 \__/¯¯\__/    Operator:           OK
 /¯¯\__/¯¯\    Envoy DaemonSet:    OK
 \__/¯¯\__/    Hubble Relay:       OK
    \__/       ClusterMesh:        disabled

DaemonSet              cilium             Desired: 8, Ready: 8/8, Available: 8/8
DaemonSet              cilium-envoy       Desired: 8, Ready: 8/8, Available: 8/8
Deployment             hubble-ui          Desired: 1, Ready: 1/1, Available: 1/1
Deployment             cilium-operator    Desired: 2, Ready: 2/2, Available: 2/2
Deployment             hubble-relay       Desired: 2, Ready: 2/2, Available: 2/2
Containers:            cilium             Running: 8
                       cilium-operator    Running: 2
                       hubble-relay       Running: 2
                       hubble-ui          Running: 1
                       cilium-envoy       Running: 8
Cluster Pods:          84/84 managed by Cilium
Helm chart version:
Image versions         cilium-envoy       quay.io/cilium/cilium-envoy:v1.28.3-31ec52ec5f2e4d28a8e19a0bfb872fa48cf7a515@sha256:bc8dcc3bc008e3a5aab98edb73a0985e6ef9469bda49d5bb3004c001c995c380: 8
                       cilium             quay.io/cilium/cilium:v1.15.5@sha256:4ce1666a73815101ec9a4d360af6c5b7f1193ab00d89b7124f8505dee147ca40: 8
                       cilium-operator    quay.io/cilium/operator-generic:v1.15.5@sha256:f5d3d19754074ca052be6aac5d1ffb1de1eb5f2d947222b5f10f6d97ad4383e8: 2
                       hubble-relay       quay.io/cilium/hubble-relay:v1.15.5@sha256:1d24b24e3477ccf9b5ad081827db635419c136a2bd84a3e60f37b26a38dd0781: 2
                       hubble-ui          quay.io/cilium/hubble-ui:v0.13.0@sha256:7d663dc16538dd6e29061abd1047013a645e6e69c115e008bee9ea9fef9a6666: 1
                       hubble-ui          quay.io/cilium/hubble-ui-backend:v0.13.0@sha256:1e7657d997c5a48253bb8dc91ecee75b63018d16ff5e5797e5af367336bc8803: 1
Errors:                cilium             cilium-7w9sw    unable to retrieve cilium status: error with exec request (pod=kube-system/cilium-7w9sw, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.3:10250, code 502: 502 Bad Gateway
                       cilium             cilium-7w9sw    unable to retrieve cilium endpoint information: error with exec request (pod=kube-system/cilium-7w9sw, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.3:10250, code 502: 502 Bad Gateway
                       cilium             cilium-kl4gl    unable to retrieve cilium status: error with exec request (pod=kube-system/cilium-kl4gl, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.6:10250, code 502: 502 Bad Gateway
                       cilium             cilium-kl4gl    unable to retrieve cilium endpoint information: error with exec request (pod=kube-system/cilium-kl4gl, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.6:10250, code 502: 502 Bad Gateway
                       cilium             cilium-rp48v    unable to retrieve cilium status: error with exec request (pod=kube-system/cilium-rp48v, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.4:10250, code 502: 502 Bad Gateway
                       cilium             cilium-rp48v    unable to retrieve cilium endpoint information: error with exec request (pod=kube-system/cilium-rp48v, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.4:10250, code 502: 502 Bad Gateway
                       cilium             cilium-shdz8    unable to retrieve cilium status: error with exec request (pod=kube-system/cilium-shdz8, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.7:10250, code 502: 502 Bad Gateway
                       cilium             cilium-shdz8    unable to retrieve cilium endpoint information: error with exec request (pod=kube-system/cilium-shdz8, container=cilium-agent): error dialing backend: proxy error from 192.168.4.2:6443 while dialing 192.168.4.7:10250, code 502: 502 Bad Gateway
```

### Kernel Version

```shell
Linux apollo 6.8.0-1004-raspi #4-Ubuntu SMP PREEMPT_DYNAMIC Sat Apr 20 02:29:55 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux
```

### Kubernetes Version

```shell
Client Version: v1.29.4+k3s1
Kustomize Version: v5.0.4-0.20230601165947-6ce0bf390ce3
Server Version: v1.29.4+k3s1
```

### Relevant Logs

K3s service logs:

```shell
May 29 19:03:42 apollo k3s[4901]: time="2024-05-29T19:03:42-04:00" level=debug msg="Wrote ping"
May 29 19:03:42 apollo k3s[4901]: time="2024-05-29T19:03:42-04:00" level=info msg="Cluster-Http-Server 2024/05/29 19:03:42 http: TLS handshake error from 192.168.4.3:49668: remote error: tls: bad certificate"
May 29 19:03:43 apollo k3s[4901]: time="2024-05-29T19:03:43-04:00" level=debug msg="cgroupv2 io stats: skipping over unmappable dbytes=0 entry"
May 29 19:03:43 apollo k3s[4901]: time="2024-05-29T19:03:43-04:00" level=debug msg="cgroupv2 io stats: skipping over unmappable dios=0 entry"
May 29 19:03:46 apollo k3s[4901]: time="2024-05-29T19:03:46-04:00" level=debug msg="cgroupv2 io stats: skipping over unmappable dbytes=0 entry"
May 29 19:03:46 apollo k3s[4901]: time="2024-05-29T19:03:46-04:00" level=debug msg="cgroupv2 io stats: skipping over unmappable dios=0 entry"
```

### Code of Conduct

- [X] I agree to follow project's Code of Conduct

**Comment by fmunteanu** [comment]
Related to https://github.com/k3s-io/k3s/issues/10252.

**Comment by fmunteanu** [comment]
Fixed in #119.
